### ISSUE TITLE
CorsHttpMethods is missing PATCH method

### ISSUE THREAD

**Issue by jiasli** [Azure/azure-storage-net#902]
https://github.com/Azure/azure-storage-net/blob/master/Lib/Common/Shared/Protocol/CorsHttpMethods.cs is missing PATCH method. 

When initializing `ServiceProperties` https://github.com/Azure/azure-storage-net/blob/07d66a4917ab03cb5e7262774235d2d2f7e65752/Lib/Common/Shared/Protocol/ServiceProperties.cs#L615

It crushes with
```
Unhandled Exception: Microsoft.Azure.Storage.StorageException: Requested value 'GET,OPTIONS,DELETE,HEAD,POST,MERGE,PUT,PATCH' was not found.
```

Sample code to reproduce:
```csharp
string storageConnectionString = "";

CloudStorageAccount storageAccount;
if (CloudStorageAccount.TryParse(storageConnectionString, out storageAccount))
{
    var myStorageClient = storageAccount.CreateCloudBlobClient();
    myStorageClient.GetServiceProperties(); // Crushes
}
```

**Comment by seanmcc-msft** [comment]
Hi,

I can repo the issue.  We'll fix it in an upcoming release.

-Sean

**Comment by seanmcc-msft** [comment]
Fixed in v11.0.0.
